phase-4-explainer.md

```diff
@@ -0,0 +1,175 @@
+  function(computedAccessibleNode) {
+    console.log(computedAccessibleNode.role);  // resolved!
+  },
+  function(error) {
+    console.log(error);  // rejected, but why?
+  });
+```
+
+The rejection case of the `Promise` is discussed in [Open Questions](#open-questions-rejection-case). Alternatively, we can use `await`:
+
+```js
+var computedAccessibleNode = await window.getComputedAccessibleNode(element);
+console.log(computedAccessibleNode.role);  // "button".
+```
+We can now access all of the computed properties exposed from one object, without relying on a leaky abstraction. This is distinct from [Phase 1](#phase-1-modifying-accessible-properties) as it would return the computed role, `"button"`, rather than `null`.  Exactly what properties a `ComputedAccessibleNode` should expose is discussed in in [Open Questions](#open-questions-accessible-properties).
+
+#### Accessibility Tree Snapshots
+`ComputedAccessibleNode`s should represent a consistent snapshot of the Accessibility Tree.
+
+#### Open Questions
+There are a number of open questions about Phase 4 that need to be discussed. This list is by no means exhaustive, but should provide a good platform for beginning discussion on the specifics of the public API design.
+
+##### 1. Accessible Properties
+The first question to answer is what properties are we expecting to expose via this API, and how? A lot of the properties are fairly self explanatory, such as `role` and `name` returning a string, or `colCount` returning an integer etc. Other attributes are not so straightforward, such as `checked`. DOM elements expose a `checked` property on all input types, which can take either of the values `true` or `false`. . Other attributes are not so straightforward, such as `checked`. Although checkboxes can only be in one of two states, checked or unchecked upon a form submission, they can take on a third visual state - `indeterminate`.
+
+```html
+<input type="checkbox" id="someCheckbox"/>
+```
+
+```js
+var checkbox = document.getElementByID("someCheckbox");
+checkbox.indeterminate = true;
+```
+This is purely a visual state, meaning that it does not impact the value of the `checked` attribute:
+
+```js
+console.log(checked.checked)  // false
+consoel.log(checked.indeterminate)  // true
+```
+
+ARIA reflects these states by allowing an author to set the checked state with `aria-checked`, assigning it to a string value of either `true`, `false`, or `mixed` (the latter reflecting the indeterminate state). This mixed checked state can be used to indicate that a parent checkbox has some nested child checkboxes that are a combination of `checked` and `unchecked`. What should a `ComputedAccessibleNode` expose these three visual states? However, as a checkbox cannot be submitted with a value of `mixed` or `indeterminate` should this be stored in a separate attribute? Three possible options are:
+1. Mimic `aria-checked`and return a string indicating which state the element is in.
```

Comment:
:+1: good discussion

Reply:
Done (and also fixed up some weird sentences)